### PR TITLE
Ctctraders 1449 add navigation to AddItems SecurityTraderDetails sub section

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: test/controllers/addItems/traderSecurityDetails/SecurityConsigneeAddressControllerSpec.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ import forms.addItems.traderSecurityDetails.SecurityConsigneeAddressFormProvider
 import matchers.JsonMatchers
 import models.reference.{Country, CountryCode}
 import models.{ConsigneeAddress, CountryList, NormalMode}
-import navigation.annotations.AddItems
+import navigation.annotations.{AddItems, TradersSecurityDetails}
```
Comment (PSouthern) [inline test/controllers/addItems/traderSecurityDetails/SecurityConsigneeAddressControllerSpec.scala:26]: Please remove the `AddItems` annotation if it's not in use.

File: test/controllers/addItems/traderSecurityDetails/SecurityConsigneeEoriControllerSpec.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ import controllers.{routes => mainRoutes}
 import forms.addItems.traderSecurityDetails.SecurityConsigneeEoriFormProvider
 import matchers.JsonMatchers
 import models.NormalMode
-import navigation.annotations.AddItems
+import navigation.annotations.{AddItems, TradersSecurityDetails}
```
Comment (PSouthern) [inline test/controllers/addItems/traderSecurityDetails/SecurityConsigneeEoriControllerSpec.scala:24]: Please remove the `AddItems` annotation if it's not in use.

File: test/controllers/addItems/traderSecurityDetails/SecurityConsigneeNameControllerSpec.scala [hunk 3]
```diff
@@ -21,7 +21,7 @@ import controllers.{routes => mainRoutes}
 import forms.addItems.traderSecurityDetails.SecurityConsigneeNameFormProvider
 import matchers.JsonMatchers
 import models.NormalMode
-import navigation.annotations.AddItems
+import navigation.annotations.{AddItems, TradersSecurityDetails}
```
Comment (PSouthern) [inline test/controllers/addItems/traderSecurityDetails/SecurityConsigneeNameControllerSpec.scala:24]: Please remove the `AddItems` annotation if it's not in use.

File: test/controllers/addItems/traderSecurityDetails/SecurityConsignorAddressControllerSpec.scala [hunk 4]
```diff
@@ -23,7 +23,7 @@ import forms.addItems.traderSecurityDetails.SecurityConsignorAddressFormProvider
 import matchers.JsonMatchers
 import models.reference.{Country, CountryCode}
 import models.{ConsignorAddress, CountryList, NormalMode}
-import navigation.annotations.AddItems
+import navigation.annotations.{AddItems, TradersSecurityDetails}
```
Comment (PSouthern) [inline test/controllers/addItems/traderSecurityDetails/SecurityConsignorAddressControllerSpec.scala:26]: Please remove the `AddItems` annotation if it's not in use.

File: test/controllers/addItems/traderSecurityDetails/SecurityConsignorEoriControllerSpec.scala [hunk 5]
```diff
@@ -21,7 +21,7 @@ import controllers.{routes => mainRoutes}
 import forms.addItems.traderSecurityDetails.SecurityConsignorEoriFormProvider
 import matchers.JsonMatchers
 import models.NormalMode
-import navigation.annotations.AddItems
+import navigation.annotations.{AddItems, TradersSecurityDetails}
```
Comment (PSouthern) [inline test/controllers/addItems/traderSecurityDetails/SecurityConsignorEoriControllerSpec.scala:24]: Please remove the `AddItems` annotation if it's not in use.

File: test/controllers/addItems/traderSecurityDetails/SecurityConsignorNameControllerSpec.scala [hunk 6]
```diff
@@ -21,7 +21,7 @@ import controllers.{routes => mainRoutes}
 import forms.addItems.traderSecurityDetails.SecurityConsignorNameFormProvider
 import matchers.JsonMatchers
 import models.NormalMode
-import navigation.annotations.AddItems
+import navigation.annotations.{AddItems, TradersSecurityDetails}
```
Comment (PSouthern) [inline test/controllers/addItems/traderSecurityDetails/SecurityConsignorNameControllerSpec.scala:24]: Please remove the `AddItems` annotation if it's not in use.

File: test/navigation/TradersSecurityDetailsNavigatorSpec.scala [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+
+    "Must go from SecurityConsignorNamePage to SecurityConsignorAddressPage" in {
+      forAll(arbitrary[UserAnswers]) {
+        answers =>
+          navigator
+            .nextPage(SecurityConsignorNamePage(index), NormalMode, answers)
+            .mustBe(routes.SecurityConsignorAddressController.onPageLoad(answers.id, index, NormalMode))
+      }
+    }
+
```
Comment (PSouthern) [inline test/navigation/TradersSecurityDetailsNavigatorSpec.scala:69]: Is there a test anywhere that skips over the Consignor pages, and straight into `AddSecurityConsigneeEoriPage` if there is a consignor for all items?

File: test/navigation/TradersSecurityDetailsNavigatorSpec.scala [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+          val updatedAnswers = answers
+            .set(AddSafetyAndSecurityConsigneePage, false)
+            .success
+            .value
+          navigator
+            .nextPage(SecurityConsignorAddressPage(index), NormalMode, updatedAnswers)
+            .mustBe(routes.AddSecurityConsigneesEoriController.onPageLoad(updatedAnswers.id, index, NormalMode))
+      }
+    }
+
```
Comment (PSouthern) [inline test/navigation/TradersSecurityDetailsNavigatorSpec.scala:82]: Is there a test anywhere that skips over the Consignor and Consignee pages, and straight into `CYA` if there is a consignor and a consignee for all items?

**Review by PSouthern — CHANGES_REQUESTED** [review]

File: test/navigation/TradersSecurityDetailsNavigatorSpec.scala (hunk 8 above)
Comment (kenneth-rayner) [reply test/navigation/TradersSecurityDetailsNavigatorSpec.scala:82]: This is in a sub ticket for the nav on the trader details section of add items CTCTRADERS-1472, there are tests and navigation from the 
'What is the consignors Eor'i number page and  'Consignors address page'
 that skip to CYA if there is only 1 consignee for all items.

File: test/navigation/TradersSecurityDetailsNavigatorSpec.scala (hunk 7 above)
Comment (kenneth-rayner) [reply test/navigation/TradersSecurityDetailsNavigatorSpec.scala:69]: This is in a sub ticket for the nav on the trader details section of add items CTCTRADERS-1472

**Review by PSouthern — APPROVED** [review]

**Pull request merged**